ambry-api/src/main/java/com/github/ambry/clustermap/ClusterParticipant.java

```diff
@@ -31,7 +32,7 @@
    * @param ambryHealthReports {@link List} of {@link AmbryHealthReport} to be registered to the participant.
    * @throws IOException
    */
-  void participate(List<AmbryHealthReport> ambryHealthReports) throws IOException;
+  void participate(List<AmbryHealthReport> ambryHealthReports, AccountService accountService) throws IOException;
```

Comment:
pass in accountService here is a little bit weird. I am think if any other solution..

Reply:
@jsjtzyy   do you have any recommendation for this? I hope we don't add AccountService to Cluster related classes.

Reply:
Let me think it through.

Reply:
Please refer to `partitionStateChangeListeners` in HelixParticipant. If we want to decouple `AccountService` from `ClusterParticipant`, we can add another type of listener, i.e `TaskCompletionListener`.

Reply:
After second thought, maybe `TaskCompletionCallback` makes more sense, because the purpose to pass in account service is to update account in ZK after the task is done and result is ready to read.

Reply:
Updated. I use callback interface and implement accountServiceCallback to decouple accountService from clusterParticipant.